lib/Parse/ParsePattern.cpp

```diff
@@ -186,9 +186,10 @@ Parser::parseParameterClause(SourceLoc &leftParenLoc,
         status |= makeParserCodeCompletionStatus();
       }
     }
-
+    SourceLoc deprecatedInoutLoc;
     // ('inout' | 'let' | 'var')?
     if (Tok.is(tok::kw_inout)) {
+      deprecatedInoutLoc = Tok.getLoc();
       param.LetVarInOutLoc = consumeToken();
       param.SpecifierKind = ParsedParameter::InOut;
     } else if (Tok.is(tok::kw_let)) {
```

Comment:
I don't think you need the deprecatedInoutLoc variable at all here.  In the cases that use it, you can just use param.LetVarInOutLoc.  In the test for "deprecatedInoutLoc.isValid()", you can just check for "param.SpecifierKind == ParsedParameter::InOut".